src/test/coins_tests.cpp

```diff
@@ -822,81 +815,83 @@ void CheckWriteCoins(const CoinEntry& parent, const CoinEntry& child, const Coin
+    CheckWriteCoins(SPENT_CLEAN,        VALUE2_DIRTY_FRESH, VALUE2_DIRTY      );
+    CheckWriteCoins(SPENT_FRESH,        VALUE2_DIRTY,       VALUE2_DIRTY_FRESH);
+    CheckWriteCoins(SPENT_FRESH,        VALUE2_DIRTY_FRESH, VALUE2_DIRTY_FRESH);
+    CheckWriteCoins(SPENT_DIRTY,        VALUE2_DIRTY,       VALUE2_DIRTY      );
+    CheckWriteCoins(SPENT_DIRTY,        VALUE2_DIRTY_FRESH, VALUE2_DIRTY      );
+    CheckWriteCoins(SPENT_DIRTY_FRESH,  VALUE2_DIRTY,       VALUE2_DIRTY_FRESH);
+    CheckWriteCoins(SPENT_DIRTY_FRESH,  VALUE2_DIRTY_FRESH, VALUE2_DIRTY_FRESH);
+
+    CheckWriteCoins(VALUE1_CLEAN,       MISSING,            VALUE1_CLEAN      );
+    CheckWriteCoins(VALUE1_FRESH,       MISSING,            VALUE1_FRESH      );
+    CheckWriteCoins(VALUE1_DIRTY,       MISSING,            VALUE1_DIRTY      );
+    CheckWriteCoins(VALUE1_DIRTY_FRESH, MISSING,            VALUE1_DIRTY_FRESH);
+    CheckWriteCoins(VALUE1_CLEAN,       SPENT_DIRTY,        SPENT_DIRTY       );
+    CheckWriteCoins(VALUE1_CLEAN,       SPENT_DIRTY_FRESH,  FAIL_NO_ENTRY     );
+    CheckWriteCoins(VALUE1_FRESH,       SPENT_DIRTY,        MISSING           );
+    CheckWriteCoins(VALUE1_FRESH,       SPENT_DIRTY_FRESH,  FAIL_NO_ENTRY     );
+    CheckWriteCoins(VALUE1_DIRTY,       SPENT_DIRTY,        SPENT_DIRTY       );
+    CheckWriteCoins(VALUE1_DIRTY,       SPENT_DIRTY_FRESH,  FAIL_NO_ENTRY     );
+    CheckWriteCoins(VALUE1_DIRTY_FRESH, SPENT_DIRTY,        MISSING           );
+    CheckWriteCoins(VALUE1_DIRTY_FRESH, SPENT_DIRTY_FRESH,  FAIL_NO_ENTRY     );
+
+    CheckWriteCoins(VALUE1_CLEAN,       VALUE2_DIRTY,       VALUE2_DIRTY      );
+    CheckWriteCoins(VALUE1_CLEAN,       VALUE2_DIRTY_FRESH, FAIL_NO_ENTRY     );
+    CheckWriteCoins(VALUE1_FRESH,       VALUE2_DIRTY,       VALUE2_DIRTY_FRESH);
+    CheckWriteCoins(VALUE1_FRESH,       VALUE2_DIRTY_FRESH, FAIL_NO_ENTRY     );
+    CheckWriteCoins(VALUE1_DIRTY,       VALUE2_DIRTY,       VALUE2_DIRTY      );
+    CheckWriteCoins(VALUE1_DIRTY,       VALUE2_DIRTY_FRESH, FAIL_NO_ENTRY     );
+    CheckWriteCoins(VALUE1_DIRTY_FRESH, VALUE2_DIRTY,       VALUE2_DIRTY_FRESH);
+    CheckWriteCoins(VALUE1_DIRTY_FRESH, VALUE2_DIRTY_FRESH, FAIL_NO_ENTRY     );
 
     // The checks above omit cases where the child flags are not DIRTY, since
     // they would be too repetitive (the parent cache is never updated in these
     // cases). The loop below covers these cases and makes sure the parent cache
     // is always left unchanged.
-    for (const CAmount parent_value : {ABSENT, SPENT, VALUE1})
-        for (const CAmount child_value : {ABSENT, SPENT, VALUE2})
-            for (const char parent_flags : parent_value == ABSENT ? ABSENT_FLAGS : FLAGS)
-                for (const char child_flags : child_value == ABSENT ? ABSENT_FLAGS : CLEAN_FLAGS)
-                    CheckWriteCoins(parent_value, child_value, parent_value, parent_flags, child_flags, parent_flags);
+    for (const CoinEntry parent : {MISSING,
```

Comment:
In commit "test: Group values and states in tests into CoinEntry wrappers" (553db5dec1334ed5e8e07e34306f16de9afc22bb)

Note: Again in this part of the test, the order cases are run in seems to be changing, but the same set of cases is run. Would be helpful to mention this in commit message.

Reply:
Done in previous step